xcdat/spatial.py

```diff
@@ -244,37 +245,36 @@ def get_weights(
         and pressure).
         """
         Bounds = TypedDict(
-            "Bounds",
-            {"domain": xr.DataArray, "region": Optional[RegionAxisBounds]},
+            "Bounds", {"weights_method": Callable, "region": Optional[np.ndarray]}
         )
+
         axis_bounds: Dict[SpatialAxis, Bounds] = {
             "X": {
-                "domain": self._dataset.bounds.get_bounds("X").copy(),
-                "region": lon_bounds,
+                "weights_method": self._get_longitude_weights,
+                "region": np.array(lon_bounds, dtype="float")
+                if lon_bounds is not None
+                else None,
             },
             "Y": {
-                "domain": self._dataset.bounds.get_bounds("Y").copy(),
-                "region": lat_bounds,
+                "weights_method": self._get_latitude_weights,
+                "region": np.array(lat_bounds, dtype="float")
+                if lat_bounds is not None
+                else None,
```

Comment:
You learn something new every day.

Reply:
Yup, they are called ternary operators (aka inline conditionals).
https://datagy.io/inline-if-in-python/

I don't use them too often for variable assignment, but it can be useful where it makes sense.